tests/parametric/test_otel_env_vars.py

```diff
@@ -0,0 +1,179 @@
+import pytest, os
+from utils import missing_feature, context, scenarios, features
+
+
+@scenarios.parametric
+@features.open_tracing_api
+class Test_Otel_Env_Vars:
+    @missing_feature(context.library < "nodejs@5.11.0", reason="Implemented in v5.11.0, v4.35.0, &v3.56.0")
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_SERVICE": "service",
+                "OTEL_SERVICE_NAME": "otel_service",
+                "DD_TRACE_LOG_LEVEL": "error",  # Node uses DD_TRACE_LOG_LEVEL
+                "DD_LOG_LEVEL": "error",
+                "DD_TRACE_DEBUG": "false",
+                "OTEL_LOG_LEVEL": "debug",
+                "DD_TRACE_SAMPLE_RATE": "0.5",
+                "OTEL_TRACES_SAMPLER": "traceidratio",
+                "OTEL_TRACES_SAMPLER_ARG": "0.1",
+                "DD_TRACE_ENABLED": "true",
+                "OTEL_TRACES_EXPORTER": "none",
+                "DD_RUNTIME_METRICS_ENABLED": "true",
+                "OTEL_METRICS_EXPORTER": "none",
+                "DD_TAGS": "foo:bar,baz:qux",
+                "OTEL_RESOURCE_ATTRIBUTES": "foo=otel_bar,baz=otel_qux",
+                "DD_TRACE_PROPAGATION_STYLE_INJECT": "b3,tracecontext",
+                "DD_TRACE_PROPAGATION_STYLE_EXTRACT": "b3,tracecontext",
+                "OTEL_PROPAGATORS": "datadog,tracecontext",
+                "DD_TRACE_OTEL_ENABLED": "false",
+                "OTEL_SDK_DISABLED": "false",
+            }
+        ],
+    )
+    def test_dd_env_var_take_precedence(self, test_agent, test_library):
+        resp = test_library.get_tracer_config()
+
+        assert resp["dd_service"] == "service"
+        assert resp["dd_log_level"] == "error"
+        assert resp["dd_trace_sample_rate"] == 0.5
+        assert resp["dd_trace_enabled"] == True
+        assert resp["dd_runtime_metrics_enabled"] == True
+        tags = resp["dd_tags"]
+        assert tags.get("foo") == "bar"
+        assert tags.get("baz") == "qux"
+        assert resp["dd_trace_propagation_style"] == "b3,tracecontext"
+        assert resp["dd_trace_debug"] == False
+        if context.library != "nodejs":
+            assert resp["dd_trace_otel_enabled"] == False  # node does not expose this variable in the config
+            assert resp["dd_trace_sample_ignore_parent"] == True  # node does not implement this env variable
```

Comment:
We should move each of these to their own test cases. I don't think any language supports `dd_trace_sample_ignore_parent` yet and and `dd_trace_otel_enabled` is not supported in ruby or go.

Reply:
OR we can just assert the otel values are not set. That could be easier and would serve as a better catch all.